### PR TITLE
lib/chkname.c: Update regex for valid names

### PR DESCRIPTION
**Pull request by alejandro-colomar**
The maximum length of 32 wasn't being enforced in the code, and POSIX doesn't specify that maximum length either, so it seems it was an arbitrary limit of the past that doesn't exist any more.  Use a regex that has no length limit.

Closes: <https://github.com/shadow-maint/shadow/issues/836>

### CODE REVIEW

**Review by hallyn — APPROVED** [review]
Looks good to me, but will wait for other approvals / complaints before merging.

**Review by ikerexxe — APPROVED** [review]
LGTM also. If you agree with the discussion in the issue I'll proceed to merging this one.

File: lib/chkname.c
Comment (vapier) [inline lib/chkname.c:35]: i'd be inclined to link to `https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html` for the definition of BRE.  googling "BRE regex" turns up docs, but POSIX is like the 7th link down the page.

Comment (alejandro-colomar) [reply lib/chkname.c:35]: I've added it to the commit message.

```
$ git range-diff shadow/master gh/validname validname 
1:  8ac64c14 ! 1:  22d753cb lib/chkname.c: Update regex for valid names
    @@ Commit message
         that has no length limit.
     
         Closes: <https://github.com/shadow-maint/shadow/issues/836>
    +    Link: <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html>
         Cc: Iker Pedrosa <ipedrosa@redhat.com>
    +    Cc: Mike Frysinger <vapier@gentoo.org>
         Signed-off-by: Alejandro Colomar <alx@kernel.org>
     
      ## lib/chkname.c ##
```

(I was even tempted to remove the comment from the source code and just put it in a commit message, to avoid it becoming stale again.)

Does that look good to you?

**Comment by alejandro-colomar** [comment]
v1b changes:

-  Rebase to master

```
$ git range-diff master..gh/validname shadow/master..validname 
1:  22d753cb = 1:  f5b4a981 lib/chkname.c: Update regex for valid names
```

**Comment by alejandro-colomar** [comment]
v1c changes:

-  Rebase to master

```
$ git range-diff master..gh/validname shadow/master..validname 
1:  f5b4a981 = 1:  e10952d5 lib/chkname.c: Update regex for valid names
```

**Review by hallyn — APPROVED** [review]

**Pull request merged**
